src/AutoRest.CSharp/Common/Generation/Writers/RestClientWriter.cs

```diff
@@ -159,13 +160,13 @@ private void WriteOperation(CodeWriter writer, RestClientMethod operation, bool
                     writer.Line($"{PipelineField}.Send({messageVariable}, cancellationToken);");
                 }
 
-                WriteStatusCodeSwitch(writer, messageVariable, operation, async);
+                WriteStatusCodeSwitch(writer, messageVariable, operation, async, returnNullOn404);
             }
             writer.Line();
         }
 
         //TODO: Do multiple status codes
-        private void WriteStatusCodeSwitch(CodeWriter writer, CodeWriterDeclaration message, RestClientMethod operation, bool async)
+        private void WriteStatusCodeSwitch(CodeWriter writer, CodeWriterDeclaration message, RestClientMethod operation, bool async, bool returnNullOn404)
```

Comment:
I don't think we need to add logic to the writer. We should add a 404 response to `RestClientMethod` (https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Common/Output/Models/Requests/RestClientMethod.cs#L33) model

Reply:
We have enough logic in the writer to generate the code you are looking for : https://github.com/Azure/autorest.csharp/blob/feature/v3/test/TestServerProjects/httpInfrastructure/Generated/MultipleResponsesRestClient.cs#L257-L258